### PR TITLE
Fixing broken links

### PR DESCRIPTION
**Pull request by svick**

### CODE REVIEW

**Comment by svick** [comment]
Other broken links I found but didn't fix:

* [Getting Started with Visual Basic](https://docs.microsoft.com/en-us/dotnet/articles/visual-basic/getting-started/) links to https://docs.microsoft.com/en-us/cpp/top/visual-cpp-in-visual-studio-2015 and https://docs.microsoft.com/en-us/office-dev/office-dev/office-and-sharepoint-development-in-visual-studio. Other articles link to cpp or office-dev too.
* [/highentropyva (C# Compiler Options)](https://docs.microsoft.com/en-us/dotnet/articles/csharp/language-reference/compiler-options/highentropyva-compiler-option) (and others) links to http://go.microsoft.com/fwlink/?LinkId=226234 which redirects to https://www.microsoft.com/download/en/details.aspx?displaylang=en&id=26788. Earlier today, it returned "502 - Web server received an invalid response while acting as a gateway or proxy server.", now it redirects to https://www.microsoft.com/library/errorpages/smarterror.aspx.
* [How to: Use Indexed Properties in COM Interop Programming (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/articles/csharp/programming-guide/interop/how-to-use-indexed-properties-in-com-interop-rogramming) links to http://go.microsoft.com/fwlink/?LinkId=179211, which redirects to http://msdnstage.redmond.corp.microsoft.com/en-us/library/microsoft.office.interop.excel.range.aspx, which returns "Server not found"
* [Walkthrough: Persisting an Object in Visual Studio (Visual Basic)](https://docs.microsoft.com/en-us/dotnet/articles/visual-basic/programming-guide/concepts/serialization/walkthrough-persisting-an-object-in-visual-studio) links to https://docs.microsoft.com/en-us/visual-studio/ide/creating-solutions-and-projects

**Comment by saldana** [comment]
### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory?branch=pr-en-us-1276) | [Details](#user-content-docs/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory.md)
[docs/core/porting/libraries.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/core/porting/libraries.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/core/porting/libraries?branch=pr-en-us-1276) | [Details](#user-content-docs/core/porting/libraries.md)
[docs/csharp/programming-guide/types/how-to-convert-a-byte-array-to-an-int.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/csharp/programming-guide/types/how-to-convert-a-byte-array-to-an-int.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/programming-guide/types/how-to-convert-a-byte-array-to-an-int?branch=pr-en-us-1276) | [Details](#user-content-docs/csharp/programming-guide/types/how-to-convert-a-byte-array-to-an-int.md)
[docs/csharp/programming-guide/types/how-to-convert-between-hexadecimal-strings-and-numeric-types.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/csharp/programming-guide/types/how-to-convert-between-hexadecimal-strings-and-numeric-types.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/programming-guide/types/how-to-convert-between-hexadecimal-strings-and-numeric-types?branch=pr-en-us-1276) | [Details](#user-content-docs/csharp/programming-guide/types/how-to-convert-between-hexadecimal-strings-and-numeric-types.md)
[docs/fsharp/tutorials/getting-started/getting-started-vscode.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/fsharp/tutorials/getting-started/getting-started-vscode.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/fsharp/tutorials/getting-started/getting-started-vscode?branch=pr-en-us-1276) | [Details](#user-content-docs/fsharp/tutorials/getting-started/getting-started-vscode.md)
[docs/fsharp/using-fsharp-on-azure/index.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/fsharp/using-fsharp-on-azure/index.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/fsharp/using-fsharp-on-azure/index?branch=pr-en-us-1276) | [Details](#user-content-docs/fsharp/using-fsharp-on-azure/index.md)
[docs/standard/base-types/best-practices-strings.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/standard/base-types/best-practices-strings.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/standard/base-types/best-practices-strings?branch=pr-en-us-1276) | [Details](#user-content-docs/standard/base-types/best-practices-strings.md)
[docs/standard/base-types/standard-numeric.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/standard/base-types/standard-numeric.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/standard/base-types/standard-numeric?branch=pr-en-us-1276) | [Details](#user-content-docs/standard/base-types/standard-numeric.md)
[docs/standard/base-types/trimming.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/standard/base-types/trimming.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/standard/base-types/trimming?branch=pr-en-us-1276) | [Details](#user-content-docs/standard/base-types/trimming.md)
[docs/standard/tour.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/standard/tour.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/standard/tour?branch=pr-en-us-1276) | [Details](#user-content-docs/standard/tour.md)
[docs/visual-basic/programming-guide/concepts/linq/linq-and-strings.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/visual-basic/programming-guide/concepts/linq/linq-and-strings.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/visual-basic/programming-guide/concepts/linq/linq-and-strings?branch=pr-en-us-1276) | [Details](#user-content-docs/visual-basic/programming-guide/concepts/linq/linq-and-strings.md)

<a name="docs/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory.md"></a>
### [docs/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory.md](https://github.com/svick/core-docs/blob/broken-links-11-2016/docs/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:Microsoft.VisualBasic.FileIO.FileSystem.GetFiles(System.String,Microsoft.VisualBasic.FileIO.SearchOption,System.String[])&quot; sourcefile=&quot;docs/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;GetFiles(String, SearchOption, String[])&lt;/a&gt;&quot;.











For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C20%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201611192336428569-1276%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ The overloads of the <xref:Microsoft.VisualBasic.FileIO.FileSystem.GetFiles%2A?d
   
 -   Use the <xref:Microsoft.VisualBasic.FileIO.FileSystem.GetFiles%28System.String%29> overload for a simple file search in a specified directory, without searching subdirectories.  
   
--   Use the [GetFiles(String, SearchOption, String\[\])](assetId:///M:Microsoft.VisualBasic.FileIO.FileSystem.GetFiles(System.String,Microsoft.VisualBasic.FileIO.SearchOption,System.String[])?qualifyHint=False&autoUpgrade=False) overload to specify additional options for your search. You can use the `wildCards` parameter to specify a search pattern. To include subdirectories in the search, set the `searchType` parameter to <xref:Microsoft.VisualBasic.FileIO.SearchOption?displayProperty=fullName>.  
+-   Use the [GetFiles(String, SearchOption, String\[\])](xref:Microsoft.VisualBasic.FileIO.FileSystem.GetFiles(System.String,Microsoft.VisualBasic.FileIO.SearchOption,System.String[])) overload to specify additional options for your search. You can use the `wildCards` parameter to specify a search pattern. To include subdirectories in the search, set the `searchType` parameter to <xref:Microsoft.VisualBasic.FileIO.SearchOption?displayProperty=fullName>.
```
Comment (svick) [inline docs/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory.md]: This is the line that's causing the warning. I don't understand why, [the method referenced here does exist](https://msdn.microsoft.com/en-us/library/ms127994(v=vs.110).aspx).

Comment (mairaw) [inline docs/visual-basic/developing-apps/programming/drives-directories-files/how-to-get-the-collection-of-files-in-a-directory.md]: We have a bug that some of these xref links are not building correctly now. That should be fixed shortly.

File: docs/visual-basic/programming-guide/concepts/linq/linq-and-strings.md [hunk 2]
```diff
@@ -35,7 +35,7 @@ LINQ can be used to query and transform strings and collections of strings. It c
  [How to: Count Occurrences of a Word in a String (LINQ) (Visual Basic)](../../../../visual-basic/programming-guide/concepts/linq/how-to-count-occurrences-of-a-word-in-a-string-linq.md)  
  Shows how to use LINQ for simple querying over text.  
   
- [How to: Query for Sentences that Contain a Specified Set of Words (LINQ) (Visual Basic)](http://msdn.microsoft.com/library/12eb44b2-f826-4864-ada0-af154e4267e9)  
+ [How to: Query for Sentences that Contain a Specified Set of Words (LINQ) (Visual Basic)](https://msdn.microsoft.com/library/mt692783.aspx)
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/concepts/linq/linq-and-strings.md]: we should link to the topic in this repo: https://github.com/dotnet/docs/blob/master/docs/visual-basic/programming-guide/concepts/linq/how-to-query-for-sentences-that-contain-a-specified-set-of-words.md

**Review by mairaw — CHANGES_REQUESTED** [review]
@svick thanks so much for fixing these! I've left two comments that would be good to address before we merge. I'll take care of the others you mention you didn't fix.

File: docs/visual-basic/programming-guide/concepts/linq/linq-and-strings.md (hunk 2 above)
Comment (svick) [inline docs/visual-basic/programming-guide/concepts/linq/linq-and-strings.md]: Done for this link and also for *How to: Generate XML from CSV Files*.

**Review by mairaw — APPROVED** [review]

**Comment by mairaw** [comment]
Thanks @svick!  I'm fixing some of the broken links you mentioned with #1281 and I'll follow up on the other two that are broken FwLinks.

**Pull request merged**